web/templates/settings/custom_user_profile_field.hbs

```diff
@@ -1,15 +1,15 @@
 <div class="custom_user_field" name="{{ field.name }}" data-field-id="{{ field.id }}">
     <span class="custom-user-field-label-wrapper {{#if field.required}}required-field-wrapper{{/if}}">
-        <label class="settings-field-label inline-block title" for="{{ field.name }}">{{ field.name }}</label>
+        <label class="settings-field-label inline-block title" for="id_custom_profile_field_input_{{ field.id }}">{{ field.name }}</label>
```

Comment:
Clicking on label does not work for user type and date type custom profile fields.

Reply:
Okay, I've added event listeners to enable them to perform some actions. I've avoided them since I didn't think my solution was great.

Reply:
We do that for the settings in group edit UI which use pills, so that might be fine. Would be good to add them in a separate commit.

Reply:
I do not see any code for that, maybe you forgot to push the code.

Reply:
Sorry, I was still in the middle of auditing all the labels and making sure that they had the correct width and had it's interactions fully functioning. I've just pushed all my changes, please have a look now.